### ISSUE TITLE
dotnet tools are not able to work with process which is running under tmux

### ISSUE THREAD

**Issue by okarpov** [dotnet/sdk#33044]
### Describe the bug
dotnet tools are not able to work with process which is running under tmux

dotnet dump collect -p 811435

Writing full to /root/sdk/core_20230605_214030
Process 811435 not running compatible .NET runtime.

Same for dotnet-gcdump and other tools.

### To Reproduce

- open SSH session
- start tmux session
- run dotnet app inside the tmux session
- run any dotnet tool with the dotnet app process id

### Exceptions (if any)

Process 811435 not running compatible .NET runtime.

### Further technical details
.NET SDK:
 Version:   7.0.302
 Commit:    990cf98a27

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.04
 OS Platform: Linux
 RID:         ubuntu.20.04-x64
 Base Path:   /usr/share/dotnet/sdk/7.0.302/

Host:
  Version:      7.0.5
  Architecture: x64
  Commit:       8042d61b17

.NET SDKs installed:
  7.0.302 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.5 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**Comment by hoyosjs** [comment]
@okarpov is this specifically about dotnet/dump?

**Comment by okarpov** [comment]
> Same for dotnet-gcdump and other tools.

as i mentioned `Same for dotnet-gcdump and other tools.`

**Comment by baronfel** [comment]
So far the tools in question are attaching to a .NET process and using the event pipes for communication? That would suggest that a runtime or configuration-level issue might need solving. Those aren't problems with .NET SDK Tools per se, more like gaps in _specific_ tools that use this API. Have you been able to replicate this kind of failure with other tools that don't use these APIs?

**Comment by hoyosjs** [comment]
Yeah, meant is this only for dotnet-dump/gcdump/counters/trace/stack and perhaps monitor?

**Comment by hoyosjs** [comment]
@baronfel if we get confirmation and I don't see it - please transfer to dotnet/diagnostics. That being said, I use tmux and hadn't noticed this...

**Comment by okarpov** [comment]
> So far the tools in question are attaching to a .NET process and using the event pipes for communication? That would suggest that a runtime or configuration-level issue might need solving. Those aren't problems with .NET SDK Tools per se, more like gaps in _specific_ tools that use this API. Have you been able to replicate this kind of failure with other tools that don't use these APIs?

this was question to me? if yes then i'm not sure what do you mean concret, please clarify

**Comment by hoyosjs** [comment]
The question was if it was only the tools in this list that repro the issue.

> Yeah, meant is this only for dotnet-dump/gcdump/counters/trace/stack and perhaps monitor?

That being said, I just tested this on Ubuntu + TMUX on .NET 7 with the same SDK and in ubuntu 20.04 and I see the following:

<img width="551" alt="image" src="https://github.com/dotnet/sdk/assets/19413848/d397c1a0-ed22-4107-8305-9472ee03b5d2">

**Comment by okarpov** [comment]
> strange, even option `ps` does not list the process in my case and says there are no dotnet processes

```# dotnet-dump ps
No supported .NET processes were found```

**Comment by okarpov** [comment]
just have realised - can it be because of this option?
```
export DOTNET_EnableDiagnostics=0 
```

**Comment by hoyosjs** [comment]
Yes, that completely turns off all diagnostic support (diagnostic pipe, profiler, debugger, etc.)
